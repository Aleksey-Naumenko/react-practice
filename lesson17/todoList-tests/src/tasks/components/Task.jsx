import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const Task = ({ text, done, id, onDeleteTask, onChangeStatus }) => {

    return (
        <li className={classNames('list-item', {'list-item_done' : done})}>
            <input 
                className="list-item__checkbox"
                defaultChecked={done}
                type="checkbox"
                onChange={() => onChangeStatus(id)}
            />
            <span className="list-item__text">{text}</span>
            <button
                className="list-item__delete-btn"
                onClick={() => onDeleteTask(id)}
            ></button>
        </li>
    );
};

export default Task;

Task.propTypes = {
    text: PropTypes.string,
    done: PropTypes.bool,
    id: PropTypes.string.isRequired,
    onDeleteTask: PropTypes.func.isRequired,
    onChangeStatus: PropTypes.func.isRequired
}