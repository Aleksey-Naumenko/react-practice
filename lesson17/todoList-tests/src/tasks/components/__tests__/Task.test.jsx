import React from 'react';
import { shallow } from 'enzyme';
import Task from '../Task';

describe('<Task />', () => {
  it ('should render TodoList', () => {
    const props = {
      onDeleteTask: jest.fn(),
      onChangeStatus: jest.fn(),
      id: 'task-id-1',
      text: 'Task-1',
    };
    const wrappedComponent = shallow(<Task {...props}/>);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it ('should render undone task', () => {
    const props = {
      onDeleteTask: jest.fn(),
      onChangeStatus: jest.fn(),
      id: 'task-id-1',
      text: 'Task-1',
      done: false
    };
    const wrappedComponent = shallow(<Task {...props}/>);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it ('should update task on checkbox checked', () => {
    const props = {
      onDeleteTask: jest.fn(),
      onChangeStatus: jest.fn(),
      id: 'task-id-1',
      text: 'Task-1',
      done: true
    };
    const wrappedComponent = shallow(<Task {...props}/>);
    wrappedComponent.find('.list-item__checkbox').simulate('change');
    expect(props.onChangeStatus).toBeCalledWith('task-id-1');
  });

  it ('should delete task', () => {
    const props = {
      onDeleteTask: jest.fn(),
      onChangeStatus: jest.fn(),
      id: 'task-id-1',
      text: 'Task-1',
      done: true
    };
    const wrappedComponent = shallow(<Task {...props}/>);
    wrappedComponent.find('.list-item__delete-btn').simulate('click');
    expect(props.onDeleteTask).toBeCalledWith('task-id-1');
  });
});