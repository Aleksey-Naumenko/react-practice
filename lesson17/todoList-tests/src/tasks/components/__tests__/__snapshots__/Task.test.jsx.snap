// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should render TodoList 1`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    Task-1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task /> should render undone task 1`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    Task-1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;
